interactive_ai/supported_models/geti_supported_models/model_manifest.py

```diff
@@ -0,0 +1,116 @@
+# Copyright (C) 2022-2025 Intel Corporation
+# LIMITED EDGE SOFTWARE DISTRIBUTION LICENSE
+
+from enum import Enum, auto
+
+from pydantic import BaseModel, Field
+
+from .hyperparameters import (
+    AugmentationParameters,
+    DatasetPreparationParameters,
+    EvaluationParameters,
+    Hyperparameters,
+    TrainingHyperParameters, EarlyStopping,
+)
+
+
+class GPUMaker(str, Enum):
+    """GPU maker names."""
+
+    NVIDIA = "nvidia"
+    INTEL = "intel"
+
+    def __str__(self) -> str:
+        """Returns the name of the GPU maker."""
+        return str(self.name)
+
+
+class ModelManifestDeprecationStatus(str, Enum):
+    """Status of a model architecture with respect to the deprecation process."""
+
+    ACTIVE = "active"  # Model architecture is fully supported, models can be trained
+    DEPRECATED = "deprecated"  # Model architecture is deprecated, models can be still viewed and trained but it's discouraged
+    OBSOLETE = "obsolete"  # Model architecture is no longer supported, models can be still viewed but not trained
+
+    def __str__(self) -> str:
+        """Returns the name of the model status."""
+        return str(self.name)
+
+
+class ModelStats(BaseModel):
+    """Information about a machine learning model."""
+
+    gigaflops: float = Field(
+        gt=0, title="Gigaflops", description="Billions of floating-point operations per second required by the model"
+    )
+    trainable_parameters: int = Field(
+        gt=0, title="Trainable parameters", description="Number of trainable parameters in the model"
+    )
+
+
+class Capabilities(BaseModel):
+    """Model capabilities configuration."""
+
+    xai: bool = Field(
+        default=False,
+        title="Explainable AI Support",
+        description="Whether the model supports explainable AI features"
+    )
+    tiling: bool = Field(
+        default=False,
+        title="Tiling Support",
+        description="Whether the model supports image tiling for processing large images"
+    )
+
+
+class ModelManifest(BaseModel):
```

Comment:
FYI later we'll add a section about the location of model weights to the model manifest. We can align offline for the details.